### PR TITLE
Add types prop to package.json

### PR DESCRIPTION
**Pull request by Masa-Shin**
Just add `"types": "index.d.ts"` to all package.json files.

### CODE REVIEW

File: packages/function-demethodize/package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
       "default": "./index.mjs"
     }
   },
+  "types": "index.d.ts",
```
Comment (angus-c) [inline packages/function-demethodize/package.json:13]: This package does not yet have a type definition file

File: packages/function-partial/package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
       "default": "./index.mjs"
     }
   },
+  "types": "index.d.ts",
```
Comment (angus-c) [inline packages/function-partial/package.json:13]: This package does not yet have a type definition file

File: packages/object-reduce/package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
       "default": "./index.mjs"
     }
   },
+  "types": "index.d.ts",
```
Comment (angus-c) [inline packages/object-reduce/package.json:13]: This package does not yet have a type definition file

**Review by angus-c — CHANGES_REQUESTED** [review]
Thanks @Masa-Shin this is a great contribution. There are just 3 packages that you added types prop for that don't yet have type definitions

File: packages/object-reduce/package.json (hunk 3 above)
Comment (angus-c) [reply packages/object-reduce/package.json:13]: Nevermind — I see it in separate PR

File: packages/function-demethodize/package.json (hunk 1 above)
Comment (Masa-Shin) [reply packages/function-demethodize/package.json]: @angus-c 
Thanks, fixed.

**Review by angus-c — APPROVED** [review]

**Comment by angus-c** [comment]
Thank you!

**Pull request merged**